.github/workflows/dockerbuild.yml

```diff
@@ -1,16 +1,50 @@
-name: Trigger Docker Build
+name: Docker Build
 
 on:
   push:
-    branches: 
-      - 'master'
+    branches:
+      - master
+  pull_request:
+  release:
+    types:
+      - created
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-       - name: Repository Dispatch
-         uses: peter-evans/repository-dispatch@v1
-         with:
-           token: ${{ secrets.DOCKER_REPO_TOKEN }}
-           repository: robertsLando/Zwave2Mqtt-docker
-           event-type: build
+      - uses: actions/checkout@v2
+
+      - name: Just build on PR
+        if: ${{ github.event_name == 'pull_request' }}
+        run: docker build .
+
+      - name: Login to dockerhub
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+      - uses: crazy-max/ghaction-docker-buildx@v1
+        if: ${{ github.event_name != 'pull_request' }}
+
+      - name: Dev
+        if: ${{ github.ref == 'git/refs/heads/master'}}
+        run: |
+          docker buildx build \
+          --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 \
+          -t ${GITHUB_REPOSITORY}:latest \
+          --push \
+          .
+
+      - name: Latest & tag
+        if: ${{ github.event_name == 'release' }}
+        run: |
+          docker buildx build \
+          --platform linux/arm64/v8,linux/amd64,linux/arm/v6,linux/arm/v7,linux/386 \
+          -t ${GITHUB_REPOSITORY}:$(echo ${GITHUB_REF} | sed "s/git\/refs\/tags\/v//") \
```

Comment:
Are you sure that `GITHUB_REPOSITORY` will be lower case and not with upper cases? 

I mean: `OpenZWave/Zwave2Mqtt` is  not valid in dockerhub, it expects: `openzwave/zwave2mqtt`

Reply:
hardcoded to avoid ambiguity